src/dotnet/ShellShim/IShellShimManager.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.ShellShim
+{
+    internal interface IShellShimManager
+    {
+        void CreateShim(FilePath targetExecutablePath, string commandName);
+
+        void RemoveShim(string commandName);
+
+        bool ShimExists(string commandName);
```

Comment:
if not have to, the caller should not need to test the existence

Reply:
Right now this is called from tests, but I'll remove from the interface and have the tests check the paths explicitly.

Reply:
Removed in upcoming push.